web/server/codechecker_server/api/report_server.py

```diff
@@ -2861,6 +2863,187 @@ def getCheckerCounts(self, run_ids, report_filter, cmp_data, limit,
                 results.append(checker_count)
         return results
 
+    @exc_to_thrift_reqfail
+    @timeit
+    def getCheckerInfo(self, run_ids, report_filter):
+        self.__require_view()
+        with DBSession(self._Session) as session:
+            max_run_histories = session.query(
+                RunHistory.run_id,
+                func.max(RunHistory.id).label('max_run_history_id'),
+            ) \
+                .filter(RunHistory.run_id.in_(run_ids) if run_ids else True) \
+                .group_by(RunHistory.run_id)
+
+            subquery = (
+                session.query(
+                    (func.string_agg(
+                        cast(Run.id, sqlalchemy.String).distinct(),
+                        ','
+                        ).label("run_id")
+                     if session.bind.dialect.name == "postgresql"
+                     else func.group_concat(Run.id.distinct()).label("run_id"))
+                    if report_filter.isUnique
+                    else Run.id.label("run_id"),
+                    Checker.id.label("checker_id"),
```

Comment:
(Style nit: Perhaps it would improve readability if these were factored out into their own variables, and then reused. Lots of indentations here, and the `X if B else Y` syntax isn't helping the readability.)